File: number3/number3.py
import sys

def input_data():
    data = sys.stdin.read().split()
    if not data:
        return False, 0, []
    try:
        amount = int(data[0])
    except ValueError:
        return False, 0, []
    if amount < 0:
        return False, 0, []
    numbers = []
    idx = 1
    for _ in range(amount):
        if idx >= len(data):
            return False, 0, []
        try:
            x = int(data[idx])
            idx += 1
        except ValueError:
            return False, 0, []
        if x <= 0:
            return False, 0, []
        numbers.append(x)
    return True, amount, numbers

def palindrome(amount, numbers):
    for i in range(amount):
        if 99 < numbers[i] < 1000:
            unit = numbers[i] % 10
            tens = numbers[i] % 100 - unit
            hung = numbers[i] % 1000 - tens - unit
            rev = (unit * 100) + tens + (hung // 100)
            if rev == numbers[i]:
                sys.stdout.write("Yes ")
            else:
                sys.stdout.write("No ")
        else:
            sys.stdout.write("- ")
    sys.stdout.write("\n")

def main():
    success, amount, numbers = input_data()
    if success:
        palindrome(amount, numbers)
    else:
        print("Введите коректное количество чисел, числа должны быть положительные")

if __name__ == "__main__":
    main()

File: number3/number3.exs
defmodule Number3 do
  def main do
    input = IO.read(:all)
    tokens = String.split(input, ~r/\s+/, trim: true) |> Enum.map(&String.to_integer/1)

    case tokens do
      [amount | rest] when amount >= 0 ->
        if length(rest) >= amount do
          numbers = Enum.take(rest, amount)
          if Enum.all?(numbers, &(&1 > 0)) do
            process_numbers(numbers)
          else
            IO.puts("Введите коректное количество чисел, числа должны быть положительные")
          end
        else
          IO.puts("Введите коректное количество чисел, числа должны быть положительные")
        end

      _ ->
        IO.puts("Введите коректное количество чисел, числа должны быть положительные")
    end
  end

  defp process_numbers(numbers) do
    Enum.each(numbers, &check_palindrome/1)
    IO.puts("")
  end

  defp check_palindrome(x) when x > 99 and x < 1000 do
    unit = rem(x, 10)
    tens = rem(x, 100) - unit
    hung = rem(x, 1000) - tens - unit
    reversed = unit * 100 + tens + div(hung, 100)

    if reversed == x do
      IO.write("Yes ")
    else
      IO.write("No ")
    end
  end

  defp check_palindrome(_) do
    IO.write("- ")
  end
end

Number3.main()
